### ISSUE TITLE
Custom callback log values are not accurately reflected in the training progress bar

### ISSUE THREAD

**Issue by TimKoornstra** [tensorflow/tensorflow#62643]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf 2.14.1

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04

### Python version

3.9, 3.10

### Current behavior?

I implemented a custom Learning Rate schedule that inherits from the `tf.keras.optimizers.schedules.LearningRateSchedule` class. I then created a custom callback to add it to the training log. However, the values shown in the progress bar do not seem to match the "actual" values. This goes for the learning rate, but also another changing variable, such as the batch number. I expected this to update correctly, but it does not seem to do so.

### Standalone code to reproduce the issue

```shell
https://colab.research.google.com/drive/1iH6AxXjVXiaWYGyTnBbpEc_FUjjCbLE4?usp=sharing
```

### Relevant log output

```shell
Epoch 1/10

End of batch 0, LR: 9.375000445288606e-06
 1/32 ━━━━━━━━━━━━━━━━━━━━ 2:16 4s/step - loss: 0.0615 - lr: 9.3750e-06 - batch: 0.0000e+00
End of batch 1, LR: 1.8750000890577212e-05

End of batch 2, LR: 2.8125001335865818e-05

End of batch 3, LR: 3.7500001781154424e-05

End of batch 4, LR: 4.6875000407453626e-05

End of batch 5, LR: 5.6250002671731636e-05
 6/32 ━━━━━━━━━━━━━━━━━━━━ 0s 12ms/step - loss: 0.0786 - lr: 3.2813e-05 - batch: 2.5000    
End of batch 6, LR: 6.562500493600965e-05

End of batch 7, LR: 7.500000356230885e-05

End of batch 8, LR: 8.437500218860805e-05

End of batch 9, LR: 9.375000081490725e-05

End of batch 10, LR: 0.00010312500671716407

End of batch 11, LR: 0.00011250000534346327
12/32 ━━━━━━━━━━━━━━━━━━━━ 0s 10ms/step - loss: 0.0834 - lr: 6.0938e-05 - batch: 5.5000
End of batch 12, LR: 0.00012187500396976247

End of batch 13, LR: 0.0001312500098720193

End of batch 14, LR: 0.0001406250084983185

End of batch 15, LR: 0.0001500000071246177

End of batch 16, LR: 0.0001593750057509169

End of batch 17, LR: 0.0001687500043772161
18/32 ━━━━━━━━━━━━━━━━━━━━ 0s 10ms/step - loss: 0.0867 - lr: 8.9063e-05 - batch: 8.5000
End of batch 18, LR: 0.0001781250030035153

End of batch 19, LR: 0.0001875000016298145

End of batch 20, LR: 0.00019687501480802894

End of batch 21, LR: 0.00020625001343432814

End of batch 22, LR: 0.00021562501206062734
23/32 ━━━━━━━━━━━━━━━━━━━━ 0s 10ms/step - loss: 0.0882 - lr: 1.1250e-04 - batch: 11.0000
End of batch 23, LR: 0.00022500001068692654

End of batch 24, LR: 0.00023437500931322575

End of batch 25, LR: 0.00024375000793952495

End of batch 26, LR: 0.00025312500656582415

End of batch 27, LR: 0.0002625000197440386
28/32 ━━━━━━━━━━━━━━━━━━━━ 0s 11ms/step - loss: 0.0890 - lr: 1.3594e-04 - batch: 13.5000
End of batch 28, LR: 0.00027187500381842256

End of batch 29, LR: 0.000281250016996637

End of batch 30, LR: 0.00029062500107102096

End of batch 31, LR: 0.0003000000142492354
32/32 ━━━━━━━━━━━━━━━━━━━━ 5s 11ms/step - loss: 0.0893 - lr: 1.5469e-04 - batch: 15.5000
```

**Comment by gah2850** [comment]
gi

**Comment by Venkat6871** [comment]
Hi **@TimKoornstra** ,
I was able to reproduce the issue on Colab using TF v2.15, and TF-nightly ,Please find the [gist](https://colab.research.google.com/gist/Venkat6871/29190e14db8cbbbe8311849fb1be9b43/62643_2-15-nightly-v.ipynb) here for reference.

Thank you!

**Comment by Venkat6871** [comment]
Hi **@TimKoornstra** ,
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999).
Thank you!
